Periodic Boundary Condition support

tbmasood opened the pull request:
Contribution:
- Added a class called PeriodicImplicitTriangulation which constructs triangulation of a grid with periodic boundary conditions.
- Modified the MorseSmaleComplex, FTMTree, TopologicalSimplification and PersistenceDiagram TTK Paraview plugins so that they have a checkbox to activate periodic boundary conditions to be used during computation.

julien-tierny commented on core/base/PeriodicImplicitTriangulation/CMakeLists.txt:
```diff
@@ -0,0 +1,5 @@
+ttk_add_base_library(periodicImplicitTriangulation
```
to comply with ttk's naming convention, please rename (git mv)
- core/base/PeriodicImplicitTriangulation/ into  core/base/periodicImplicitTriangulation/
thanks for changing that in the involved cmake files.

tbmasood replied:
Done.

julien-tierny commented on core/base/triangulation/Triangulation.h:
```diff
@@ -3100,18 +3102,40 @@ namespace ttk {
+      int ret = periodicImplicitTriangulation_.setInputGrid(
+        xOrigin, yOrigin, zOrigin, xSpacing, ySpacing, zSpacing, xDim, yDim,
+        zDim);
+      int retPeriodic = implicitTriangulation_.setInputGrid(
```
it seems that the variables "ret" and "retPeriodic" need to be switched.

tbmasood replied:
Corrected in latest commit.

julien-tierny commented on core/base/PeriodicImplicitTriangulation/PeriodicImplicitTriangulation.h:
```diff
@@ -0,0 +1,13204 @@
+#ifndef _PERIODICIMPLICITTRIANGULATION_H
```
stupid question: does **all** the API support periodicity? (or just a selection of the functions)

tbmasood replied:
All functions are re-written to support periodicity.

tbmasood commented:
In the latest commits, I have fixed the minor changes requested. Further, I have rewritten the code to make it more concise, reducing the lines of code to about a third of the original.

julien-tierny commented:
alright, that looks ready to go.
cheers!
julien
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

On Monday, 16 September 2019 13:52:19 CEST tbmasood wrote:
> @tbmasood requested your review on: topology-tool-kit/ttk#268 Periodic Boundary Condition support.
> 
>
